src/poetry/puzzle/provider.py

```diff
@@ -125,7 +125,7 @@ def __init__(
         self,
         package: Package,
         pool: Pool,
-        io: Any,
+        io: NullIO,
```

Comment:
I'm reasonably sure this should be a `cleo.io.IO` -- unless for some reason we never pass a real IO object in?

Reply:
I agree, I guess mypy doesn't notice because cleo isn't exporting its types